internal/engine/integration/deployment_test.go

```diff
@@ -228,7 +228,7 @@ func Test_Deployment(t *testing.T) {
 				TxID:   "test",
 			})
 			if tc.err != nil {
-				require.ErrorIs(t, err, tc.err)
+				require.ErrorAs(t, err, &tc.err)
```

Comment:
This will pass any `tc.err` value because it's just a regular `error err` and anything can be put in there.

The semantics of `errors.Is` that make sure `tc.err` has the correct error *value* to that in `err` is what we need.  The problem is that `CombineParseErrors` untypes any errors that were in the slice. `ParseErrors` `Err` method is a related issue as it really shouldn't exist since `ParseErrors` itself is an `error`.  The missing piece is implementing either `Unwrap() []error` or `Unwrap() error` for `ParseErrors`.

I'll suggest a revision shortly.

Reply:
There's a little more to change than expected.  There are two listeners that need modification.  But also to keep wrapped errors detectable, `ParseError.Err` needs to become an `error` instead of a `string`, and then methods like this need to take an `error`:

```go
func (e *ErrorListener) NodeErr(node *Node, errType ParseErrorType, msg string) {
	e.Errs = append(e.Errs, &ParseError{
		ParserName: e.name,
		Type:       errType,
		Err:        msg,
		Node:       e.adjustNode(node),
	})
}
```

Because here's where the error's identity gets thrown out:

```go
t.errs.NodeErr(p0.GetNode(), types.ParseErrorTypeSemantic, fmt.Sprintf(`%s: "%s"`, types.ErrUnknownForeignProcedure.Error(), p0.Name))
```

Looking for a neat resolution, but I think every call to `NodeErr` also needs to change.

Reply:
Pushed a resolution: https://github.com/kwilteam/kwil-db/pull/698/commits/820f6b6255f04b3e68182e30c842ad4cf9d122ca
Feel free to pop that off if you want to go a different way.  I think it's best to wrap the errors.

Reply:
I appreciate the help here, thanks. Took a look at your commit, will keep it.

I always assumed that error typing was based on whether the error message contained the correct string.